cmd/tempo/app/modules.go

```diff
@@ -306,6 +315,48 @@ func (t *App) initMemberlistKV() (services.Service, error) {
 	return t.MemberlistKV, nil
 }
 
+func (t *App) initUsageReport() (services.Service, error) {
+	if !t.cfg.UsageReport.Enabled {
+		return nil, nil
+	}
+
+	t.cfg.UsageReport.Leader = false
+	if t.isModuleActive(Ingester) {
+		t.cfg.UsageReport.Leader = true
+	}
+
+	usagestats.Target(t.cfg.Target)
+
+	var err error
+	var reader backend.RawReader
+	var writer backend.RawWriter
+
+	switch t.cfg.StorageConfig.Trace.Backend {
```

Comment:
A couple things come to mind.  Having a client instance dedicated to this module seems like a good way to avoid contention/blocking on client use in other parts of the system.  I don't know this to be the case with our clients, but just something I've seen.  Also, since all of the main components will be running this code that relies on the backend, it didn't seem right to require all components to require the ingester just to get the client.

Reply:
So, tempodb should have no issues with concurrency. However, I do see now that the standard tempodb interfaces do not support what you want and we would not want to adjust them for this feature.  I also see now that we'd have to start up a storage module for the distributor, which we definitely would not want.

Agree this is likely the best way to do things.

I do think a small improvement would be to move this code to the `/tempodb/backend` package which we call like:

`backend.New(t.cfg.StorageConfig.Trace) (reader, writer, compactor, err)`

it could then be reused by `tempodb.New()`

Reply:
I like this idea of centralizing the New for the backend client creation.  At first pass, this creates a dependency cycle, but perhaps after I address some of the other comments here we might be able to get out of the loop.

Reply:
Yeah, let's refactor the backend later.